parser-typechecker/src/Unison/PrintError.hs

```diff
@@ -281,7 +281,7 @@ renderTypeError e env src curPath = case e of
         debugSummary note
       ]
   FunctionApplication {..} ->
-    let fte = Type.removePureEffects ft
+    let fte = Type.removePureEffects False ft
```

Comment:
Boolean blindness here; recommend a newtype

Reply:
I didn't entirely agree on this one, so leaving it as is. My thinking is: if you saw `DontKeepEmptied` here instead of `False`, are you going to know what that means without consulting the docs on `removePureEffects`? Adding a newtype which is literally only used in one function like this feels overkill, and it's another level of indirection to follow (to understand it, you now need to look at the docs on the type, and then see how it's used by the function - easier to just look at the docs on the function).

The places I'd newtype a boolean are if the concept crops up again and again, or perhaps if I've got a function that takes two or more booleans and it'd otherwise be error prone keeping them straight. Or perhaps if it's a very public facing function (though I'd probably introduce a pair of functions with a common implementation before newtyping a boolean for just the one case).